person.rb

```diff
@@ -1,24 +1,28 @@
-# frozen_string_literal: true
+# Person class
+require_relative 'nameable'
 
-class Person
-  def initialize(age, _parent_permission = true, name = 'Unknown')
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, parent_permission: true, name: 'Unknown')
+    super(name)
     @id = Random.rand(1..1000)
-    @name = name
     @age = age
+    @parent_permission = parent_permission
   end
 
-  # for both getters and setters
-  attr_accessor :id
-  attr_accessor :name
-  attr_accessor :age
-
   def can_use_services?
     is_of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
-  def is_of_age?
+  def of_age?
     @age >= 18
   end
 end
```

Comment:
### NOTICE:
To make sure that your application is working as expected, I recommend you run the below commands for testing it before asking a new review
```
person = Person.new(22, 'maximilianus')
 person.correct_name
 capitalizedPerson = CapitalizeDecorator.new(person)
 capitalizedPerson.correct_name
 capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
 capitalizedTrimmedPerson.correct_name
```